horace_core/spaghetti_plot.m

```diff
@@ -644,3 +644,15 @@ function plot_labels(labels,xvals)
 set(gca, 'XTickLabel', labels);
 
 end
+
+%-----------------------------------------------------------------------
+function out = set_argout(nout, varargin)
+% set output arguments depending on number of output arguments
+% requested.
+%
+% expected usage is to set varargout, and because of this, no
+% output cell array is allocated (is this reasonable?)
+for i=1:nout
+    out{i} = varargin{i};
+end
+end
```

Comment:
This does not feel comfortable. You should force the use of varargout, or provide a mechanism to swtich according to use. Suggest following the use of nargout(@function_handle) where the return value is negative if varargout is used.

Reply:
Ah... my laziness exposed. I got rid of a call to this function in many other places, but just copied the original into this function as the last bastion of a piece of code that also made me uncomfortable. Mea culpa and I am properly chastised. I'll get rid of it altogether in the revised PR.